include/triton/Dialect/Triton/IR/Dialect.h

```diff
@@ -27,6 +27,10 @@ struct GlobalMemory : public SideEffects::Resource::Base<GlobalMemory> {
   StringRef getName() final { return "<GlobalMemory>"; }
 };
 
+struct SharedMemory : public SideEffects::Resource::Base<GlobalMemory> {
+  StringRef getName() final { return "<GlobalMemory>"; }
```

Comment:
It's weird to me if shared memory effects are inherited from global memory. 
1. Does shared memory mean GPU shared memory or something else?
2. If the former, exposing it in TritonOps seems like breaking our semantics because TritonOps are device agnostic.
3. If the later, can we change a name?
4. Also, why shared memory's `getName` method returns `<GlobalMemory>`?

Reply:
Whoops, this was copy-pasted from `GlobalMemory` definition above and I forgot to completely fixup.

> Does shared memory mean GPU shared memory or something else?

Yes this is GPU shared memory.

> If the former, exposing it in TritonOps seems like breaking our semantics because TritonOps are device agnostic.

I could try to come up with some way to hide the fact these are using shared memory, but at the end of the day TMA descriptors are already a highly device specific, so I'm not sure there's any benefit there.

Reply:
And I should add that the triton language level doesn't actually see the shared memory, it only exists in the frontend when building the IR.

Reply:
> I could try to come up with some way to hide the fact these are using shared memory, but at the end of the day TMA descriptors are already highly device specific, so I'm not sure there's any benefit there.

Got it